neuro_san/internals/run_context/langchain/langchain_run_context.py

```diff
@@ -604,6 +604,14 @@ def parse_tool_output(self, tool_output: Dict[str, Any]) -> List[BaseMessage]:
         tool_message = AgentToolResultMessage(content=tool_result_dict.get("content"),
                                               tool_result_origin=tool_output.get("origin"))
 
+        # Integrate any sly data
+        tool_sly_data: Dict[str, Any] = tool_output.get("sly_data")
+        if tool_sly_data is not None and tool_sly_data != self.tool_caller.sly_data and bool(tool_sly_data):
+            # We have sly data from the tool output that is not the same as our own
```

Comment:
tool_sly_data is a Python dictionary (pointer). Is it possible that tool_sly_data==self.tool_sly_data,
but content has been modified?

Reply:
I don't think so, in either case for ==.